src/checkers/inference/VariableAnnotator.java

```diff
@@ -558,33 +558,27 @@ public VariableSlot addImpliedPrimaryVariable(AnnotatedTypeMirror atm, final Ann
     }
 
     /**
-     * Create and store constraints from pre-annotated code and implicits from the underlying type system.
+     * Given an atm, replace its real annotation from pre-annotated code and implicit from the underlying type system
+     * by the equivalent varAnnotation, or creating a new VarAnnotation for it if doesn't have any existing annotations.
      */
-    private VariableSlot createEquivalentSlotConstraints(AnnotatedTypeMirror atm, Tree tree, final AnnotationLocation location) {
+    private VariableSlot replaceOrCreateEquivalentVarAnno(AnnotatedTypeMirror atm, Tree tree, final AnnotationLocation location) {
         VariableSlot varSlot = null;
-        Slot constantSlot = null;
         AnnotationMirror realQualifier = null;
 
-        // Create constraints for pre-annotated code and constant slots when the variable slot is created.
         AnnotationMirror existinVar = atm.getAnnotationInHierarchy(varAnnot);
-        if (!atm.getAnnotations().isEmpty()) {
+        if (existinVar != null) {
+            varSlot = slotManager.getVariableSlot(atm);
+        } else if (!atm.getAnnotations().isEmpty()) {
             realQualifier = atm.getAnnotationInHierarchy(realTop);
             if (realQualifier == null) {
                 atm.addAnnotation(realTop);
```

Comment:
Rasing an error here. It is not reasonable to have a qual neither from VarAnno, and RealQual.